2017 Robot/src/org/usfirst/frc/team4999/lights/NeoPixels.java

```
@@ -25,103 +29,30 @@ public static NeoPixels getInstance() {
-			strip.writeBulk(buffer); 
-			
-			*/
-			/*
-			strip.write(1, START_FRAME);
-			byte[] color = new byte[3];
-			for(int i = 0; i < currentState.length; i++) {
-				color[0] = (byte) safe(currentState[i].getRed());
-				color[1] = (byte) safe(currentState[i].getGreen());
-				color[2] = (byte) safe(currentState[i].getBlue());
-				strip.writeBulk(color);
-			}
-			strip.write(1, SHOW_FRAME);
-			*/
-			/*
-			ByteBuffer buffer = ByteBuffer.allocateDirect(2 + currentState.length * 3);
-			buffer.put((byte) START_FRAME);
-			for(int i = 0; i < currentState.length; i++ ) {
-				buffer.put((byte) safe(currentState[i].getRed()));
-				buffer.put((byte) safe(currentState[i].getGreen()));
-				buffer.put((byte) safe(currentState[i].getBlue()));
-			}
-			buffer.put((byte) SHOW_FRAME);
-			
-			strip.writeBulk(buffer, buffer.capacity());*/
-			millis = System.currentTimeMillis();
-			int[] buffer;
-			if((currentState.length * 3) % 2 == 0) {
-				buffer = new int[(currentState.length * 3)];
-			} else {
-				buffer = new int[(currentState.length * 3) + 1];
-				buffer[(currentState.length * 3)] = 1;
-			}
-			
-			strip.write(1, START_FRAME);
-			int j = 0;
-			for(int i = 0; i < currentState.length; i++ ) {
-				buffer[j++] = safe(currentState[i].getRed());
-				buffer[j++] = safe(currentState[i].getGreen());
-				buffer[j++] = safe(currentState[i].getBlue());
-			}
-			
-			for(int i = 0; i < buffer.length; ) {
-				strip.write(buffer[i++], buffer[i++]);
-			}
-			
-			strip.write(1, SHOW_FRAME);
-			
-			System.out.println("WriteTime: " + (System.currentTimeMillis() - millis));
+			// Show the sent packets
+			strip.writeBulk(Packet.showPacket().fillBuffer(), Packet.showPacket().getPacketSize());
 			
+			return (System.currentTimeMillis() - millis);
 		} catch (Exception e) {
-			System.out.println(e.getMessage());
-			System.out.println(e.getStackTrace());
+			// The generic try-catch prevents an error in the purely cosmetic neopixels from killing the whole robot
```

It should be better to throw a Checked exception instead of returning -1 since I'm not convinced that this method should be returning anything.